docs/proposals/platform-extension.md

```diff
@@ -0,0 +1,105 @@
+# Platform Extension
+**Author**: Marcin Franczyk, Sascha Haase, Moritz Bracht, Sankalp Rangare
+
+**Status**: Draft proposal.
+
+## Goals
+
+This proposal is to enhance the current ADDON/extension mechanism and introduce a new platform-extension mechanism which will provide easier way to manage the lifecycle of extensions.
```

Comment:
Maybe at some point we could replace the addon mechanism completely. Let's see how the idea of platform extensions will evolve.